File: PathSum.cpp
#include<iostream>
#include<vector>

using namespace std;

struct TreeNode{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    bool hasPathSum(TreeNode* root, int sum) {
        
        if(root==NULL) return false;
        if(root->left==NULL && root->right==NULL) return sum==root->val;
        
        return hasPathSum(root->left, sum-root->val) || hasPathSum(root->right, sum-root->val);
    }
};

int main(){
    
    TreeNode *root=new TreeNode(5);
    TreeNode *a=new TreeNode(4);
    TreeNode *b=new TreeNode(8);
    TreeNode *c=new TreeNode(11);
    TreeNode *d=new TreeNode(13);
    TreeNode *e=new TreeNode(4);
    TreeNode *f=new TreeNode(7);
    TreeNode *g=new TreeNode(2);
    TreeNode *h=new TreeNode(1);
    
    root->left=a;
    root->right=b;
    a->left=c;
    b->left=d;
    b->right=e;
    c->left=f;
    c->right=g;
    e->right=h;
    
    int sum;
    cin>>sum;
    
    Solution *solution=new Solution();
    solution->hasPathSum(root, sum) ? cout<<"True" : cout<<"False";
    
    return 0;
    
}

File: PathSum.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def hasPathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if root==None:
            return False
        if root.left==None and root.right==None:
            return sum==root.val
        return self.hasPathSum(root.left, sum-root.val) or self.hasPathSum(root.right, sum-root.val)
        
if __name__=='__main__':
    root=TreeNode(5);
    a=TreeNode(4);
    b=TreeNode(8);
    c=TreeNode(11);
    d=TreeNode(13);
    e=TreeNode(4);
    f=TreeNode(7);
    g=TreeNode(2);
    h=TreeNode(1);
    
    root.left=a;
    root.right=b;
    a.left=c;
    b.left=d;
    b.right=e;
    c.left=f;
    c.right=g;
    e.right=h;
    
    sum = int(raw_input())
    
    solution = Solution()
    print(solution.hasPathSum(root, sum))
